Provide private image registry for cap-operator (#7977)
kyma-project/test-infra: configs/terraform/modules/artifact-registry/README.md

# gcp-artifact-registry-terraform

Provide private image registry in GCP

## Usage

> Configure variables in `terraform.tfvars` (configs/terraform/environments/prod/terraform.tfvars) file

Reviewer: ```suggestion
Configure variables in the [`terraform.tfvars`](../configs/terraform/environments/prod/terraform.tfvars) file.
```

> Default values

Reviewer: ```suggestion
These are the default values:
```

> Configure variables in [`terraform.tfvars`](../../environments/prod/terraform.tfvars) file.

Reviewer: ```suggestion
Configure variables in the [`terraform.tfvars`](../../environments/prod/terraform.tfvars) file.
```

Reviewer: Please also fix the broken link.

```terraform
artifact_registry_owner          = "neighbors"
artifact_registry_type           = "development"
immutable_artifact_registry      = false
artifact_registry_multi_region   = true
artifact_registry_primary_area  = "europe"
```

> Note:

Reviewer: ```suggestion
When you use the GCP private image registry, consider the following: 
```

> - Soluton is prepared for GCP Service Account related execution

Reviewer: ```suggestion
- The solution is prepared for GCP Service Account related execution
```

> - The solution is prepared for GCP Service Account related execution.

Reviewer: ```suggestion
- The solution is prepared for the GCP Service Account related execution.
```

> - `roles/artifactregistry.write` role binding is part of the solution ([Artifact Registry Repository Access Control](https://cloud.google.com/artifact-registry/docs/access-control))

Reviewer: ```suggestion
- The `roles/artifactregistry.write` role binding is part of the solution. To learn more. read [Artifact Registry Repository Access Control](https://cloud.google.com/artifact-registry/docs/access-control)).
```

> - Vulnerability scanning is on by default

Reviewer: ```suggestion
- Vulnerability scanning is enabled by default.
```

> - You must define `artifact_registry_serviceaccount` in tfvars file.

Reviewer: ```suggestion
- You must define `artifact_registry_serviceaccount` in the .tfvars file.
```

> - `roles/artifactregistry.writer` role binding is part of the solution. To learn more, read [Artifact Registry Repository Access Control](https://cloud.google.com/artifact-registry/docs/access-control).

Reviewer: ```suggestion
- The `roles/artifactregistry.writer` role binding is part of the solution. To learn more, read [Artifact Registry Repository Access Control](https://cloud.google.com/artifact-registry/docs/access-control).
```

# gcp-artifact-registry-terraform

Provide private image registry in GCP

## Usage

Configure variables in `terraform.tfvars` (configs/terraform/environments/prod/terraform.tfvars) file

Default values

> ```terraform
> artifact_registry_module         = "cap-operator"
> artifact_registry_owner          = "neighbors"
> artifact_registry_type           = "development"
> immutable_artifact_registry      = false
> artifact_registry_multi_region   = true
> ```

Reviewer 2: ```suggestion
```terraform
artifact_registry_owner          = "neighbors"
artifact_registry_type           = "development"
immutable_artifact_registry      = false
artifact_registry_multi_region   = true
artifact_registry_primary_area  = "europe"
```
```

artifact_registry_type           = "development"
immutable_artifact_registry      = false
artifact_registry_multi_region   = true
```

Note:
- Soluton is prepared for GCP Service Account related execution

> - Credential file is the service account key file in json format

Reviewer 2: ```suggestion
```
We don't use any credentials file.
